tests/bluetooth/bsim_bt/bsim_test_adv_resume/src/bs_bt_utils.c

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bs_bt_utils.h"
+#include "argparse.h"
+#include "bs_pc_backchannel.h"
+
+BUILD_ASSERT(CONFIG_BT_MAX_PAIRED >= 2, "CONFIG_BT_MAX_PAIRED is too small.");
+BUILD_ASSERT(CONFIG_BT_ID_MAX >= 2, "CONFIG_BT_ID_MAX is too small.");
+
+#define BS_SECONDS(dur_sec)    ((bs_time_t)dur_sec * 1000000)
+#define TEST_TIMEOUT_SIMULATED BS_SECONDS(60)
+
+void test_tick(bs_time_t HW_device_time)
+{
+	bs_trace_debug_time(0, "Simulation ends now.\n");
+	if (bst_result != Passed) {
+		bst_result = Failed;
+		bs_trace_error("Test did not pass before simulation ended.\n");
+	}
+}
+
+void test_init(void)
+{
+	bst_ticker_set_next_tick_absolute(TEST_TIMEOUT_SIMULATED);
+	bst_result = In_progress;
+}
+
+DEFINE_FLAG(flag_is_connected);
+
+void wait_connected(void)
+{
+	UNSET_FLAG(flag_is_connected);
+	WAIT_FOR_FLAG(flag_is_connected);
+	printk("connected\n");
+}
+
+void wait_disconnected(void)
+{
+	SET_FLAG(flag_is_connected);
+	WAIT_FOR_FLAG_UNSET(flag_is_connected);
+	printk("disconnected\n");
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	UNSET_FLAG(flag_is_connected);
+}
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	if (err != 0) {
+		return;
+	}
+
+	SET_FLAG(flag_is_connected);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+void bs_bt_utils_setup(void)
+{
+	int err;
+
+	err = bt_enable(NULL);
+	ASSERT(!err, "bt_enable failed.\n");
+}
+
+static bt_addr_le_t last_scanned_addr;
+
+static void scan_connect_to_first_result_device_found(const bt_addr_le_t *addr, int8_t rssi,
+						      uint8_t type, struct net_buf_simple *ad)
+{
+	struct bt_conn *conn;
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	int err;
+
+	/* We're only interested in connectable events */
+	if (type != BT_HCI_ADV_IND && type != BT_HCI_ADV_DIRECT_IND) {
+		FAIL("Unexpected advertisement type.");
+	}
+
+	bt_addr_le_to_str(addr, addr_str, sizeof(addr_str));
+	printk("Got scan result, connecting.. dst %s, RSSI %d\n", addr_str, rssi);
+
+	err = bt_le_scan_stop();
+	ASSERT(!err, "Err bt_le_scan_stop %d", err);
+
+	err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, BT_LE_CONN_PARAM_DEFAULT, &conn);
```

Comment:
Consider making it this
```suggestion
	err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, BT_LE_CONN_PARAM_DEFAULT, 
				&conn);
```
as 80 is the still the recommended maximum line length. I see a few other lines here and there that could be made to less than 80.

Reply:
yeah. It would be great to have a different level than error in checkpatch. That way we could keep 80chars as the default, but still allow for the newer 100 (or is it 120) for select cases. I usually just run clang-format.